src/ctype/CType.ts

```diff
@@ -57,7 +60,23 @@ export default class CType implements ICType {
     CTypeUtils.errorCheck(cTypeInput)
     this.schema = cTypeInput.schema
     this.owner = cTypeInput.owner
-    this.hash = cTypeInput.hash
+
+    if (!cTypeInput.hash) {
```

Comment:
I think this is handled via the `CTypeUtils.errorCheck` and the new `CType.fromSchema` functions